modules/programs/powerline-go.nix

```diff
@@ -0,0 +1,271 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.powerline-go;
+
+  moduleNames = [
+    "aws"
+    "cwd"
+    "docker"
+    "docker-context"
+    "dotenv"
+    "duration"
+    "exit"
+    "git"
+    "gitlite"
+    "hg"
+    "host"
+    "jobs"
+    "kube"
+    "load"
+    "newline"
+    "nix-shell"
+    "node"
+    "perlbrew"
+    "perms"
+    "plenv"
+    "root"
+    "shell-var"
+    "shenv"
+    "ssh"
+    "svn"
+    "termtitle"
+    "terraform-workspace"
+    "time"
+    "user"
+    "venv"
+    "vgo"
+  ];
+
+  optionSpecs = [
```

Comment:
Could we not use an attribute-set for this, so we don't have to keep track of the upstream options?

Reply:
I don't understand what you mean. Can you be a bit more specific?

Reply:
Do you want something completely generic so home-manager users could type all key-value pairs they want and those would be written as is in the configuration file (without any type check or documentation)? Something like https://rycee.gitlab.io/home-manager/options.html#opt-programs.starship.settings?

I don't have any strong opinion. I have seen both solutions in home-manager. It would be good that home-manager maintainers decide what they want so everyone could follow the same best practice.

Reply:
Yes, I mean a generic option.
`nixpkgs` goes for generic options without type checking and just a link to the documentation (at least in the PRs I did). But I don't know how @rycee wants to have it.

Reply:
I discussed with @rycee and he agrees with you. He would like me to follow [some best practice](https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#valuable-options).

Reply:
fixed